sqlite.go

```diff
@@ -0,0 +1,542 @@
+// Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package sqlite implements a database/sql driver for SQLite3.
+//
+// This driver requires a file: URI always be used to open a database.
+// For details see https://sqlite.org/c3ref/open.html#urifilenames.
+//
+//
+// Memory Mode
+//
+// In-memory databases are popular for tests.
+// If opening a database in memory mode with the "?mode=memory" query
+// parameter, make sure immediatley after opening to set
+//
+//	db.SetMaxOpenConns(1)
+//
+// This driver does not support shared cache mode, as the SQLite
+// authors recommend against it:
+//
+//	"Any use of shared cache is discouraged."
+//	https://www.sqlite.org/c3ref/enable_shared_cache.html
+//
+// For far more realistic in-memory database testing, use
+//
+//	PRAGMA journal_mode=WAL;
+//	PRAGMA synchronous=OFF;
+//
+// On any modern OS, this will generate syscalls as part of writing
+// data, but will never block on I/O.
+//
+//
+// Binding Types
+//
+// SQLite is flexible about type conversions, and so is this driver.
+// Almost all "basic" Go types (int, float64, string) are accepted and
+// directly mapped into SQLite, even if they are named Go types.
+// The time.Time type is also accepted (described below).
+// Unrecognized types are tested to be implementations of
```

Comment:
"are tested to be" made my brain skip a gear and have to go back. How about:

Values that implement encoding.TextMarshaler or json.Marshaler are stored in SQLite in their marshaled form.